src/lucky/error_handler.cr

```diff
@@ -34,8 +34,18 @@ class Lucky::ErrorHandler
   end
 
   private def call_error_action(context : HTTP::Server::Context, error : Exception) : HTTP::Server::Context
-    context.response.status_code = 500
+    status_code = status_code_by_error(error)
+    context.response.status_code = status_code
     action.new(context).perform_action(error)
     context
   end
+
+  private def status_code_by_error(error : Exception)
+    case error
+    when Lucky::Exceptions::InvalidParam
+      Lucky::Action::Status::UnprocessableEntity.value
+    else
+      Lucky::Action::Status::InternalServerError.value
+    end
```

Comment:
What if instead of using conditionals we add a module that can be mixed in to errors. Then any error can implement it and set a status code. Here's roughly what I mean:

```crystal
module Lucky::DefaultErrorResponse
  abstract def default_lucky_status_code : Int32
end
```

Then an error can implement it:

```crystal
class Lucky::Exceptions::InvalidParam
  include Lucky::DefaultErrorResponse
  
  def default_lucky_status_code
    422
  end
end
```

In the handler you could do something like this:

```
private def status_code_by_error(error: Exception)
  Lucky::Action::Status::InternalServerError.value
end

private def status_code_by_error(error: Lucky::DefaultErrorResponse)
  error.default_lucky_status_code
end
```

That way we can easily add status code for other errors, or shards can add this too w/o us needing to change Lucky. What do you think? Does that make sense?

Also I don't love the names I suggested :P If you have a better idea definitely feel free to change my suggestion!

Reply:
Interesting! I like that better. Let me take a look.

Reply:
Cool! Another idea for the module might be `Lucky::HasDefaultErrorResponse`